ejb/ejb2-connector/ejb2-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/custom/connector/EJBUtil.java

```diff
@@ -0,0 +1,193 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.custom.connector;
+
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.axis2.databinding.typemapping.SimpleTypeMapper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.mediators.Value;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+
+import javax.ejb.EJBHome;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.rmi.PortableRemoteObject;
+import javax.xml.stream.XMLStreamException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+public class EJBUtil {
+    private static final Log log = LogFactory.getLog(EJBUtil.class);
+
+    /**
+     * @param instance instance of an ejb object
+     * @param method   method that we want to call
+     * @param args     arguments object for method
+     * @return invoke arguments into method and return the value which actual method returns
+     */
+    public static Object invokeInstanceMethod(Object instance, Method method, Object[] args) {
+        Class[] paramTypes = method.getParameterTypes();
+        if (paramTypes.length != args.length) {
+            handleException("Provided argument count does not match method the "
+                    + "parameter count of method '" + method.getName() + "'. Argument count = "
+                    + args.length + ", method parameter count = " + paramTypes.length);
+        }
+        Object[] processedArgs = new Object[paramTypes.length];
+        for (int i = 0; i < paramTypes.length; ++i) {
+            if (args[i] == null || paramTypes[i].isAssignableFrom(args[i].getClass())) {
+                processedArgs[i] = args[i];
+            } else if (SimpleTypeMapper.isSimpleType(paramTypes[i])) {
+                try {
+                    processedArgs[i] = SimpleTypeMapper.getSimpleTypeObject(paramTypes[i],
+                            AXIOMUtil.stringToOM("<a>" + args[i] + "</a>"));
+                } catch (XMLStreamException e) {
+                    handleException("XMLStreamException ", e);
+                }
+            } else {
+                handleException("Incompatible argument found in " + i + "th argument "
+                        + "for '" + method.getName() + "' method.");
+            }
+        }
+        try {
+            return method.invoke(instance, processedArgs);
+        } catch (IllegalAccessException e) {
```

Comment:
Since we are using java 1.7 we can include both in one catch eg: IllegalAccessException | InvocationTargetException. Also better catch Exception at the end?